app/subscribers/super_good/solidus_taxjar/spree/reporting_subscriber.rb

```diff
@@ -3,9 +3,12 @@ module SolidusTaxjar
     module Spree
       module ReportingSubscriber
         include ::Spree::Event::Subscriber
+        include SolidusSupport::LegacyEventCompat::Subscriber
 
         if ::Spree::Event.method_defined?(:register)
           ::Spree::Event.register("shipment_shipped")
+        elsif ("::Spree::Bus".constantize rescue nil)
```

Comment:
I don't like this, and we may be able to do this in a better way. One thing that would be cool is if the `SolidusSupport::LegacyEventCompat::Bus` class had a `#register` method. Right now it only supports `#publish`.

Reply:
```suggestion
        elsif class_available?("::Spree::Bus")
        
# ...

def class_available?(class_name)
  # ...
end
```

Maybe not worth doing, but moving the constantize check into a private method would make the control flow easier to skim.

Reply:
After reading through the [update instructions](https://github.com/solidusio/solidus/pull/4342) on the PR that added Omnes, I think the better way to register this is in the initializer which is already conditional on the Solidus verison.